src/HealthChecks.AzureApplicationInsights/AzureApplicationInsightsHealthCheck.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HealthChecks.AzureApplicationInsights
+{
+    public class AzureApplicationInsightsHealthCheck : IHealthCheck
+    {
+        //from https://docs.microsoft.com/en-us/azure/azure-monitor/app/ip-addresses#outgoing-ports
+        private readonly string[] m_appInsightsUrls =
+            {
+            "https://dc.applicationinsights.azure.com",
+            "https://dc.applicationinsights.microsoft.com",
+            "https://dc.services.visualstudio.com"
+            };
+        private readonly string _instrumentationKey;
+
+        public AzureApplicationInsightsHealthCheck(string instrumentationKey)
+        {
+            if (string.IsNullOrWhiteSpace(instrumentationKey))
+            {
+                throw new ArgumentNullException(nameof(instrumentationKey));
+            }
+
+            _instrumentationKey = instrumentationKey;         
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                return await CheckHealthAsyncInternal(context, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+            }
+        }
+        
+        private async Task<HealthCheckResult> CheckHealthAsyncInternal(HealthCheckContext context, CancellationToken cancellationToken = default)
```

Comment:
I would inline this method into `CheckHealthAsync`.

Reply:
done